core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/JoinClause.kt

```diff
@@ -65,7 +67,40 @@ sealed class SelectJoinClause : ISelectJoinClause {
     constructor(joinType: JoinType, bucket: Bucket, onKey: Field<out ValidType>, forBucket: Bucket, parentClause: ISelectFromClause) {
         this.onType = ON_KEY_FOR
         this.joinType = joinType
-        this.bucket = bucket
+        this.resolvable = bucket
+        this.onKey = onKey
+        this.forBucket = forBucket
+        this.parentClause = parentClause
+        this.onCondition = null
+        this.onKeys = null
+    }
+
+    constructor(joinType: JoinType, bucket: AliasedBucket, onCondition: TypeExpression<BooleanType>, parentClause: ISelectFromClause) {
```

Comment:
I think it's cleaner if you just call the other constructor with changing one argument

Reply:
done